Спринт 15. Домашнее задание

Yarogi opened the pull request.

Yarogi committed:
Спринт 15.1 Добавлены пакеты

Yarogi committed:
Спринт 15.2 Добавлен интерфейс и реализация медиа-плеера

Yarogi committed:
Спринт 15.3 - Плеер выведен в интерфейс для дальнейшего переподключения

Yarogi committed:
Спринт 15.4 - Добавлен интерактор

Yarogi committed:
Спринт 15.5 - Search - Retrofit clean-arch

Yarogi committed:
Спринт 15.5 - Search - sharedPreference clean-arch

Yarogi committed:
Merge pull request #24 from Yarogi/dev12_Search

Спинт 15. Бонусное задание. Activity Search как Clean Architecture

Yarogi committed:
Merge pull request #25 from Yarogi/dev11

Спринт 15. Домашнее задание. Clean

i-masloed commented on app/src/main/java/com/example/playlistmaker/data/impl/TracksRepositoryImpl.kt:
```diff
@@ -0,0 +1,28 @@
+            val data = (response as TrackResponse).results.map { TrackFromTrackDtoMapper.get(it) }
+            return Resource.Success(data)
+        } else {
+            return Resource.Error("A network error has occurred")
```
Рекомендация: строки было бы здорово вынести в ресурсы

Yarogi replied:
Поправил

Yarogi committed:
1. Исправлен баг с паузой (остановкой трека)
2. Строка с ошибкой сети добавлена в ресурс
3. Исправлен метод реагирования на код ответа

Yarogi committed:
Исправлен баг с отображением View

Yarogi committed:
Merge pull request #27 from Yarogi/dev11

Корректировка багов

Yarogi commented:
Поправил баг с тем, что трек воспроизводится при свернутом окне(+ немного косметики по замечаниям)
По второму замечанию - не понял о чем речь. Я некое запаздание паузы обнаружил только, если управление жестами включено. На кнопках навигации задержки с паузой вроде нет.
